src/arg_parsers/identify_format.rs

```diff
@@ -0,0 +1,225 @@
+use crate::arg_parse_err::ArgParseErr;
+use std::ffi::OsStr;
+
+// https://imagemagick.org/script/escape.php
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Var {
+    Colorspace,
+    CurrentImageHeightInPixels,
+    CurrentImageWidthInPixels,
+    ImageDepth,
+    ImageFileFormat,
+    ImageFilename,
+    LayerCanvasPageGeometry,
+    MagickFilename,
+    OriginalImageSize,
+    PageCanvasHeight,
+    PageCanvasWidth,
+    PageCanvasXOffset,
+    PageCanvasYOffset,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Token {
+    Literal(String),
+    Whitespace(usize),
+    Var(Var),
+}
+
+#[derive(Debug, Clone, PartialEq, Default)]
+pub struct IdentifyFormat {
+    pub template: Option<Vec<Token>>,
+}
+
+enum ParseState {
+    Initial,
+    Literal,
+    Whitespace,
+    Var,
+}
+
+impl TryFrom<&std::ffi::OsStr> for IdentifyFormat {
+    type Error = ArgParseErr;
+
+    fn try_from(s: &OsStr) -> Result<Self, Self::Error> {
+        let mut tokens: Vec<Token> = Vec::new();
+        let ascii = s.as_encoded_bytes();
```

Comment:
Not 100% we are not losing something here again.

> The byte encoding is an unspecified, platform-specific, self-synchronizing superset of UTF-8. By being a self-synchronizing superset of UTF-8, this encoding is also a superset of 7-bit ASCII.

This reads a bit cryptic to me, not sure what to make of it.

> Note: As the encoding is unspecified, any sub-slice of bytes that is not valid UTF-8 should be treated as opaque and only comparable within the same Rust version built for the same target platform.

We're safe here, we only compare within the same program.